Facing build errors

After switching to this cmake script of cryptopp, I'm facing the following include error:
![image](https://user-images.githubusercontent.com/47210458/187026529-fae0f732-b539-4e15-9653-d27879af53f1.png)

I noticed, you still have `cryptopp` in your `vcpkg.json` file (which pulls the old cryptopp-cmake) and you have `cryptopp` (the new one) in your externals. Could that be the problem?

I tried to build your commit on Windows, and I face problems with vcpkg and fmtlib before I compete the configure...

```
-- Performing Test FMT_HAS_VARIANT - Success
-- Required features: cxx_variadic_templates
-- Looking for _strtod_l
-- Looking for _strtod_l - found
CMake Error at externals/vcpkg/scripts/buildsystems/vcpkg.cmake:618 (_add_library):
  _add_library cannot create ALIAS target "fmt::fmt" because another target
  with the same name already exists.
Call Stack (most recent call first):
  externals/dynarmic/externals/fmt/CMakeLists.txt:240 (add_library)


CMake Error at externals/vcpkg/scripts/buildsystems/vcpkg.cmake:618 (_add_library):
  _add_library cannot create ALIAS target "fmt::fmt-header-only" because
  another target with the same name already exists.
Call Stack (most recent call first):
  externals/dynarmic/externals/fmt/CMakeLists.txt:296 (add_library)

```

I am running into the same issue on Windows. These are my build steps:

1. mkdir build
2. cd build
3. cmake ..
4. cmake --build .

Here's the [CMakeCache.txt](https://github.com/abdes/cryptopp-cmake/files/9453345/CMakeCache.txt)

Output of `cmake --build .`:
```
C:\Users\Admin\Source\Repos\cryptopp-cmake\build>cmake ..
-- Building for: Visual Studio 17 2022
=> Project : cryptopp v8.7.0 (c7fa777bb92d)
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22622.
-- The CXX compiler identification is MSVC 19.33.31629.0
-- The C compiler identification is MSVC 19.33.31629.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.33.31629/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.33.31629/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.37.0.windows.1")
-- downloading/updating cryptopp
-- cryptopp directory not found, cloning...
Cloning into 'C:/Users/Admin/Source/Repos/cryptopp-cmake/build/_deps/cryptopp'...
--
Switched to a new branch 'tag_CRYPTOPP_8_7_0'
--
-- Crypto++ auto fetched at: C:/Users/Admin/Source/Repos/cryptopp-cmake/build/_deps/cryptopp
-- Adding install integration test: int-install-default
-- Adding install integration test: int-install-prefix
=> Module : cryptopp
-- [cryptopp] CMake version 3.23.1
-- [cryptopp] System Windows
-- [cryptopp] Processor AMD64
-- [cryptopp] Target architecture detected as: x86_64 -> CRYPTOPP_AMD64
-- Looking for C++ include winapifamily.h
-- Looking for C++ include winapifamily.h - found
-- The ASM_MASM compiler identification is MSVC
-- Found assembler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.33.31629/bin/Hostx64/x64/ml64.exe
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- [cryptopp] Generating cmake package config files
-- [cryptopp] Generating pkgconfig files
-- [cryptopp] Platform: x86_64
-- [cryptopp] Compiler definitions:  CRYPTOPP_DATA_DIR="C:/Users/Admin/Source/Repos/cryptopp-cmake/build/_deps/cryptopp";_WIN32_WINNT=0x0A00
-- [cryptopp] Compiler options: /DWIN32 /D_WINDOWS /W3 /GR /EHsc  /FIwinapifamily.h;/GR;/EHsc
-- [cryptopp] Build type:
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Admin/Source/Repos/cryptopp-cmake/build

C:\Users\Admin\Source\Repos\cryptopp-cmake\build>cmake --build .
MSBuild version 17.3.0+f67e3d35e for .NET Framework
  Checking Build System
  Building Custom Rule C:/Users/Admin/Source/Repos/cryptopp-cmake/cryptopp/CMakeLists.txt
  Assembling C:\Users\Admin\Source\Repos\cryptopp-cmake\build\_deps\cryptopp\rdrand.asm...
C:\Users\Admin\Source\Repos\cryptopp-cmake\build\_deps\cryptopp\rdrand.asm(1): fatal error A1000: cannot open file : cr
yptopp.dir\Debug\/__/_deps/cryptopp/rdrand.asm.obj [C:\Users\Admin\Source\Repos\cryptopp-cmake\build\cryptopp\cryptopp.
vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\BuildCustomizations\masm.targets(70,5
): error MSB3721: Der Befehl "ml64.exe /c /nologo /Zi /Fo"cryptopp.dir\Debug\/__/_deps/cryptopp/rdrand.asm.obj" /D"WIN3
2" /D"_WINDOWS" /D"CMAKE_INTDIR="Debug"" /D"_M_X64" /I "C:\Users\Admin\Source\Repos\cryptopp-cmake\build\_deps\cryptopp
" /I "C:\Users\Admin\Source\Repos\cryptopp-cmake\build\_deps" /W3 /errorReport:prompt  /Ta"C:\Users\Admin\Source\Repos\
cryptopp-cmake\build\_deps\cryptopp\rdrand.asm"" wurde mit Code 1 beendet. [C:\Users\Admin\Source\Repos\cryptopp-cmake\
build\cryptopp\cryptopp.vcxproj]
```

--- a/cryptopp/CMakeLists.txt
+++ b/cryptopp/CMakeLists.txt
@@ -405,10 +405,12 @@ if(MSVC AND NOT DISABLE_ASM)
     enable_language(ASM_MASM)
     if(NOT DISABLE_RDRAND)
       list(APPEND cryptopp_SOURCES_ASM ${CRYPTOPP_PROJECT_DIR}/rdrand.asm)
-      # workaround https://github.com/abdes/cryptopp-cmake/issues/13
-      set_source_files_properties(${CRYPTOPP_PROJECT_DIR}/rdrand.asm
-                                  PROPERTIES COMPILE_OPTIONS
-                                             "/Fo\$(IntDir)rdrand.asm.obj")
+      if(${CMAKE_GENERATOR} MATCHES "Visual Studio.*")
+        # workaround https://github.com/abdes/cryptopp-cmake/issues/13
+        set_source_files_properties(${CRYPTOPP_PROJECT_DIR}/rdrand.asm
+                                    PROPERTIES COMPILE_OPTIONS
+                                               "/Fo\$(IntDir)rdrand.asm.obj")
+      endif()
     endif()
     if(NOT DISABLE_RDSEED)
       list(APPEND cryptopp_SOURCES_ASM ${CRYPTOPP_PROJECT_DIR}/rdseed.asm)

--- a/cryptopp/CMakeLists.txt
+++ b/cryptopp/CMakeLists.txt
@@ -405,6 +405,10 @@ if(MSVC AND NOT DISABLE_ASM)
     enable_language(ASM_MASM)
     if(NOT DISABLE_RDRAND)
       list(APPEND cryptopp_SOURCES_ASM ${CRYPTOPP_PROJECT_DIR}/rdrand.asm)
+      # workaround https://github.com/abdes/cryptopp-cmake/issues/13
+      set_source_files_properties(${CRYPTOPP_PROJECT_DIR}/rdrand.asm
+                                  PROPERTIES COMPILE_OPTIONS
+                                             "/Fo\$(IntDir)rdrand.asm.obj")
     endif()
     if(NOT DISABLE_RDSEED)
       list(APPEND cryptopp_SOURCES_ASM ${CRYPTOPP_PROJECT_DIR}/rdseed.asm)

